src/app/model/User.js

```diff
@@ -39,6 +39,24 @@ class User {
       });
   }
 
+  linkPhoneAuthentication(firebase, phoneNumber, recaptchaVerfier, callback) {
+    return firebase
+      .auth()
+      .currentUser.linkWithPhoneNumber(phoneNumber, recaptchaVerfier)
+      .then(function (confirmationResult) {
+        var code = window.prompt("Provide your SMS code");
+        recaptchaVerfier.clear();
+        return confirmationResult.confirm(code).then(() => {
+          callback();
+        });
+      })
+      .catch((err) =>
+        alert(
+          "You already have an account associated with this phone number. Please sign in using that number."
```

Comment:
can this situation even happen? 
User have an account -> he is logged in 
If he is logged in, and have phone number associate with the account -> he has a phone number

How can we come to this alert?